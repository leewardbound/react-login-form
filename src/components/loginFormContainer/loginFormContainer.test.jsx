import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import LoginFormContainer from '.';

const mockStore = configureMockStore([ thunk ]);

describe('LoginFormContainer', function () {
  it('handle the unauthenticated case correctly', function () {
    const store = mockStore({ auth: { authenticated: false }});

    const component = renderer.create(
      <LoginFormContainer store={store} title="foo">
        <h1>Lorem ipsum</h1>
      </LoginFormContainer>
    );

    expect(component).toMatchSnapshot();
  });

  it('handle the authenticated case correctly', function () {
    const store = mockStore({ auth: { authenticated: true }});

    const component = renderer.create(
      <LoginFormContainer store={store} title="foo">
        <h1>Lorem ipsum</h1>
      </LoginFormContainer>
    );

    expect(component).toMatchSnapshot();
  });
});
