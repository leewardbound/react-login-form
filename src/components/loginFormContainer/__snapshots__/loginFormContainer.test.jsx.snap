// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormContainer handle the authenticated case correctly 1`] = `
<p
  className="fJygQn"
  type="success"
>
  Login successful
</p>
`;

exports[`LoginFormContainer handle the unauthenticated case correctly 1`] = `
<div
  className="enZFBb"
>
  <h1
    className="OEbrL"
  >
    <span>
      foo
    </span>
  </h1>
  <h1>
    Lorem ipsum
  </h1>
</div>
`;
